src/components/Todo/Todo.js

```diff
@@ -4,25 +4,23 @@ import Checkbox from 'components/Checkbox/Checkbox'
 import { FaTrashAlt } from 'react-icons/fa'
 import style from './todo.module.scss'
 
-const Todo = ({ todo, index, handleDeleteIconClick }) => {
+const Todo = ({ todo, index, handleDeleteIconClick, onStatusChange, status }) => {
   const [visible, setVisible] = useState(false)
   const handleIconDelete = () => setVisible(!visible)
```

Comment:
Acá no estás actuando sobre el ícono, por lo que podrías llamarlo:

```suggestion
  const handleHover = () => setVisible(!visible)
```